backend/test/authenticate.test.js

```diff
@@ -0,0 +1,43 @@
+import { AuthenticateMiddleware } from '../src/middlewares/authenticate';
+import { SessionService } from '../src/services/sessionService';
+
+jest.mock('../src/services/sessionService');
+
+const valid_input = { headers: { authorization: 'Bearer Some.JWT.5-Token' } };
+const invalid_input = { headers: { authorization: '**********' } };
+
+describe('Testing authenticate middleware', () => {
+  it('It should fail with 401 Authorization token is missing', () => {
+    const authenticateMiddleware = new AuthenticateMiddleware();
+    const next = jest.fn(x => console.log('next was called'));
+    const mockResponse = () => {
+      const res = {};
+      res.status = 401;
+      res.json = 'hi';
+      return res;
+    };
+
+    const resp = mockResponse();
+    authenticateMiddleware.authenticate(invalid_input, resp, next);
+    expect(resp.status).toBe(401);
```

Comment:
@LLaszlo89 
- here you don't need to mock the `next` that won't be used, you can just pass a `null`
- `resp` on the other hand need to contain a `status()` that will be _called_ and that should be checked
- also the thing that `status()` is returning should have a `json()` that will be _called_ and that also should be checked